crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -0,0 +1,312 @@
+    // but the correctness of the note script could not be guaranteed.
+    Maybe,
+}
+
+// The enum holding all possible account interfaces which could be loaded to some account.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub enum AccountComponentInterface {
+    BasicWallet,
+    BasicFungibleFaucet,
+    // Internal value of the `RpoFalcon512` holds the storage offset where the private key is
+    // stored
+    RpoFalcon512(u8),
+    Custom(Vec<AccountProcedureInfo>),
+}
+
+impl AccountComponentInterface {
+    /// Creates a set of [AccountComponentInterface] instances, specifying components which were
+    /// used to create an account with the provided procedures array.
+    pub fn from_procedures(procedures: &[AccountProcedureInfo]) -> BTreeSet<Self> {
+        let rpo_falcon_procs = rpo_falcon_512_library()
+            .mast_forest()
+            .procedure_digests()
+            .collect::<Vec<Digest>>();
+
+        debug_assert!(rpo_falcon_procs.len() == 1);
+        let rpo_falcon_proc = rpo_falcon_procs[0];
+
+        let mut component_interface_set = BTreeSet::new();
+
+        if basic_wallet_library().mast_forest().procedure_digests().all(|proc_digest| {
+            procedures.iter().any(|proc_info| proc_info.mast_root() == &proc_digest)
+        }) {
+            component_interface_set.insert(AccountComponentInterface::BasicWallet);
+        }
+
+        if basic_fungible_faucet_library()
+            .mast_forest()
+            .procedure_digests()
+            .all(|proc_digest| {
+                procedures.iter().any(|proc_info| proc_info.mast_root() == &proc_digest)
+            })
+        {
+            component_interface_set.insert(AccountComponentInterface::BasicFungibleFaucet);
+        }
+
+        procedures.iter().for_each(|proc_info| {
+            if proc_info.mast_root() == &rpo_falcon_proc {
+                component_interface_set
+                    .insert(AccountComponentInterface::RpoFalcon512(proc_info.storage_offset()));
+            }
+        });
+
+        let custom_interface_procs = procedures
+            .iter()
+            .filter(|proc_digest| {
+                !basic_wallet_library()
+                    .mast_forest()
+                    .procedure_digests()
+                    .any(|wallet_proc_digest| &wallet_proc_digest == proc_digest.mast_root())
+                    && !basic_fungible_faucet_library()
+                        .mast_forest()
+                        .procedure_digests()
+                        .any(|faucet_proc_digest| &faucet_proc_digest == proc_digest.mast_root())
+                    && (&rpo_falcon_proc != proc_digest.mast_root())
+            })
+            .cloned()
+            .collect::<Vec<AccountProcedureInfo>>();
+
+        if !custom_interface_procs.is_empty() {
+            component_interface_set
+                .insert(AccountComponentInterface::Custom(custom_interface_procs));
+        }
```

Comment:
This may be fine for now, but originally I was thinking that we'd try to "group" related procedures together. For example, if we have 3 procedures:
```
foo (offset = 1, size = 2)
bar (offset = 1, size = 2)
baz (offset = 3, size = 1)
```
We'd end up with the following interfaces:
```rust
[Custom(vec![foo, bar]), Custom(vec![baz])]
```